roadmap/implementers-guide/src/node/collators/collator-protocol.md

```diff
@@ -0,0 +1,133 @@
+
+### Validators
+
+On the validator side of the protocol, validators need to accept incoming connections from collators. They should keep some peer slots open for accepting new speculative connections from collators and should disconnect from collators who are not relevant.
+
+```dot process
+digraph G {
+  label = "Declaring, advertising, and providing collations";
+  labelloc = "t";
+  rankdir = LR;
+
+  subgraph cluster_collator {
+      rank = min;
+      label = "Collator";
+      graph[style = border, rank = min];
+
+      c1, c2 [label = ""];
+  }
+
+  subgraph cluster_validator {
+      rank = same;
+      label = "Validator";
+      graph[style = border];
+
+      v1, v2 [label = ""];
+  }
+
+  c1 -> v1 [label = "Declare and advertise"];
+
+  v1 -> c2 [label = "Request"];
+
+  c2 -> v2 [label = "Provide"];
+
+  v2 -> v2 [label = "Note Good/Bad"];
+}
+```
+
+When peers connect to us, they can `Declare` that they represent a collator with given public key. Once they've declared that, they can begin to send advertisements of collations. The peers should not send us any advertisements for collations that are on a relay-parent outside of our view.
+
+The protocol tracks advertisements received and the source of the advertisement. The advertisement source is the `PeerId` of the peer who sent the message. We accept one advertisement per collator per source per relay-parent.
+
+As a validator, we will handle requests from other subsystems to fetch a collation on a specific `ParaId` and relay-parent. These requests are made with the [`CollatorProtocolMessage`][CPM]`::FetchCollation`. To do so, we need to first check if we have already gathered a collation on that `ParaId` and relay-parent. If not, we need to select one of the advertisements and issue a request for it. If we've already issued a request, we shouldn't issue another one until the first has returned.
+
+When acting on an advertisement, we issue a `WireMessage::RequestCollation`. If the request times out, we need to note the collator as being unreliable and reduce its priority relative to other collators. And then make another request - repeat until we get a response or the chain has moved on.
+
+As a validator, once the collation has been fetched some other subsystem will inspect and do deeper validation of the collation. The subsystem will report to this subsystem with a [`CollatorProtocolMessage`][CPM]`::ReportCollator` or `NoteGoodCollation` message. In that case, if we are connected directly to the collator, we apply a cost to the `PeerId` associated with the collator and potentially disconnect or blacklist it.
+
```

Comment:
As mentioned in my other comment on PeerSet, validators should also be connected to a few other validators on the same parachain, and forward received collations onto them. (The details of neighbour selection are mentioned in #1348)

Other validators are more trusted than collators, so the protocol wire message here only needs to consist of the actual `Collation`, but if it's simpler to re-use the collator-validator wire message for now, it wouldn't hurt. However in the future once we get onto passing around whitelists/blacklists of collators, the message types would have to diverge.

Reply:
> As mentioned in my other comment on PeerSet, validators should also be connected to a few other validators on the same parachain, and forward received collations onto them

Agreed, but this is handled by the PoV distribution and Statement distribution subsystems, not collation distribution.